### PR TITLE
Not sharing the absolute Deadline object.

### PR DESCRIPTION
**Pull request by dparmar**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by stefanobaghino-da — COMMENTED** [review]
Thanks for correctly pointing out that `withDeadline` is still not actually part of a stable release and apologies for the mistake, for some reason it appeared on a stale change log for a v1.16 snapshot (you can see it mentioned in #10522 but the mention there is stale).

Since we're still in time to change the signature, I'd recommend we switch the method name as well to make sure we use the correct terminology, since we're not using an absolute deadline but a relative timeout.

Please also mention this in a changelog tag in a commit body. You can read how this works in CONTRIBUTING.md.

Please note that this cannot be merged until you accept the CLA.

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala [hunk 1]
```diff
@@ -36,7 +37,7 @@ class CommandSubmissionClientImplTest
   it should "timeout after deadline exceeded" in {
```
Comment (stefanobaghino-da) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala:37]: There is still no test checking that the error is fixed.

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImpl.java [hunk 2]
```diff
@@ -24,18 +24,18 @@
 public class CommandSubmissionClientImpl implements CommandSubmissionClient {
 
   private final String ledgerId;
-  private final CommandSubmissionServiceGrpc.CommandSubmissionServiceFutureStub serviceStub;
+  private CommandSubmissionServiceGrpc.CommandSubmissionServiceFutureStub serviceStub;
```
Comment (stefanobaghino-da) [inline language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImpl.java:27]:
```suggestion
  private final CommandSubmissionServiceGrpc.CommandSubmissionServiceFutureStub serviceStub;
```
There's no actual need of making this `final`, forking a stub is cheap.

**Review by stefanobaghino-da — APPROVED** [review]
Thanks for taking care of the comments.

Remember to sign the CLA so that we can merge this.

**Comment by stefanobaghino-da** [comment]
> Thanks for taking care of the comments.
> 
> Remember to sign the CLA so that we can merge this.

Apologies, I see you signed it already, I was looking at a stale GitHub page.

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala (hunk 1 above)
Comment (dparmar) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala:37]: I think a good test would be to make 2 command submissions on the same instance of CommandSubmissionClientImpl. 

1. The 1st (time > timeout) should fail (similar to existing deadline test case).
2. The 2nd (time < timeout) should pass OK.

However, I am completely new to Scala, so would like to seek your help.

Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala:37]: I opened a PR on your repo [here](https://github.com/dparmar/daml/pull/1). If you have a look at that and merge it, those changes should be reflected here (for some reason the code formatting tool made the diff larger than I expected, apologies for that).

Comment (dparmar) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImplTest.scala]: Thanks a lot for your help! I just merged it.

**Pull request merged**